src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.spring.intro.dao.impl;
+
+import com.spring.intro.dao.UserDao;
+import com.spring.intro.exeptions.DataProcessingException;
+import com.spring.intro.model.User;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import lombok.extern.log4j.Log4j;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@Log4j
+public class UserDaoImpl implements UserDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
```

Comment:
What is better: setter, constructor, or field injection?

Reply:
constructor